### DISCUSSION TITLE
Valheim server experiencing lag

### DISCUSSION THREAD

**Discussion by HugoKamps** [GameServerManagers/LinuxGSM#3273] [Q&A] [answered]
I'm hosting a dedicated Valheim server using LinuxGSM on a server rented with DigitalOcean and am experiencing a good amount of lag when playing with around 6+ players. The server is located in Amsterdam (we're all from the Netherlands).

The lag occurs at these moments:
- Whenever someone joins the server
- One or more people are travelling far away
- Lots of people are cutting down trees near each other

The lag symptoms:
- Items can't be picked up
- Doors can't be opened
- The world loads really slow when travelling (like sailing into invisible rocks)
- Trees disappear and then reappear when chopping them down

Before we played on a server with 4GB of RAM, but we decided to upgrade to see if this solved the issue. Unfortunately it didn't, as the server's RAM usage didn't increase (before it was around 48%, now with double the memory it's 24%).

Is this a known issue or is this related to our server? And is there any way to improve this? 

Thanks in advance :)

Hardware / Distribution specs: 
Distro Details
=================================
Distro:    Ubuntu 20.04.1 LTS
Arch:      x86_64
Kernel:    5.4.0-65-generic
Hostname:  valheim-server
Uptime:    0d, 11h, 4m
tmux:      tmux 3.0a
glibc:     2.31

Server Resource
=================================
CPU
Model:      Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz
Cores:      4
Frequency:  2294.608MHz
Avg Load:   0.44, 0.29, 0.21

Memory
Mem:       total  used   free   cached  available
Physical:  7.8GB  2.1GB  5.6GB  2.7GB   5.6GB
Swap:      0B     0B     0B

Storage
Filesystem:  /dev/vda1
Total:       78G
Used:        6.5G
Available:   71G

Network
IP:           0.0.0.0
Internet IP:  -----------

Game Server Resource Usage
=================================
CPU Used:  18.2%
Mem Used:  24%    1916MB

Storage
Total:        4.4G
Serverfiles:  998M
Backups:      3.0G

**Comment by dgibbs64** [comment]
Well your hardware stats and resource usage look good to me, doesn't look overloaded. Since Valheim is a new game server it could be just bugs etc that need ironing out.  Hopefully a few people who have been playing Valheim will come forward and say if they have experienced the same

**Comment by wldwalker** [comment, marked as answer]
Quite a few people have posted in the past(on multiple forums, including Digital Oceans own forums) of having issues with services that use UDP when hosting with DigitalOcean. A few people in the LGSM discord have been reporting very odd behavior with their Valheim server when they use Digital Ocean (and Linode for that matter). It is not ideal and makes little sense what Digital Ocean is doing but people have switched to hosts like Vultr and have had no problems. Your resource usage looks perfectly normal.

**Comment by rconjoe** [comment]
A majority of users deploying Valheim on DigitalOcean have reported similar issues. Most of the time, actually, players are unable to connect to the instance at all. It does appear to be something wrong with DO's own internal routing and UDP packets not making it where they're supposed to. 

The only solution for now is using a different vps provider.
